modules/gnome/nixos.nix

```diff
@@ -2,38 +2,46 @@
   lib,
   pkgs,
   config,
+  mkTarget,
   ...
 }:
+mkTarget {
+  name = "gnome";
+  humanName = "GNOME and GDM";
+  autoEnable =
+    config.services.xserver.desktopManager.gnome.enable
+    || config.services.xserver.displayManager.gdm.enable;
```

Comment:
`mkTarget` currently isn't covered by the `mkEnabkeTargetWith` PR, so migrating modules with dynamic `autoEnable` to `mkTarget` is kinda blocked for now.

Reply:
I've updated #1244 to also add `autoEnableExpr` support to `mkTarget`, so these kinda migrations are no longer blocked.